File: Basics/Hash-Map/hash.go
package main

import "fmt"

type HashMap map[string]string

func (h HashMap) insert(key, value string) {
	h[key] = value
}

func (h HashMap) retrieve(key string) string {
	return h[key]
}

func main() {
	myHashMap := make(HashMap)
	myHashMap.insert("key1", "value1")
	myHashMap.insert("key2", "value2")

	fmt.Println(myHashMap.retrieve("key1")) // Output: value1
	fmt.Println(myHashMap.retrieve("key2")) // Output: value2
}

File: Basics/Hash-Map/hash.rs
use std::collections::HashMap;

fn main() {
    let mut my_hash_map = HashMap::new();
    
    my_hash_map.insert("key1", "value1");
    my_hash_map.insert("key2", "value2");

    println!("{:?}", my_hash_map.get("key1")); // Output: Some("value1")
    println!("{:?}", my_hash_map.get("key2")); // Output: Some("value2")
}
